clean_build.sh

```diff
@@ -1,32 +1,47 @@
 #!/bin/sh
+
 cd "$(dirname "$0")" || exit 1
+
 # some environments have cmake v2 as 'cmake' and v3 as 'cmake3'
 # check for cmake3 first then fallback to just cmake
-B_CMAKE=`type cmake3 2>/dev/null`
-if [ $? -eq 0 ]; then
-    B_CMAKE=`echo "$B_CMAKE" | cut -d' ' -f3`
+
+B_CMAKE=$(command -v cmake3 2>/dev/null)
+if [ "$?" -eq 0 ]; then
+    continue
 else
-    B_CMAKE=cmake
+    if command -v cmake 2>/dev/null; then
+        B_CMAKE=$(command -v cmake)
+    else
+        echo "ERROR: CMake not in $PATH, cannot build! Please install CMake, or if this persists, file a bug report."
+        exit 1
+    fi
 fi
-# default build configuration
-B_BUILD_TYPE=${B_BUILD_TYPE:-Debug}
+
+B_BUILD_TYPE="${B_BUILD_TYPE:-Debug}"
+B_CMAKE_FLAGS="-DCMAKE_BUILD_TYPE=${B_BUILD_TYPE} ${B_CMAKE_FLAGS:-}"
+
 if [ "$(uname)" = "Darwin" ]; then
-    # OSX needs a lot of extra help, poor thing
-    # run the osx_environment.sh script to fix paths
-    . ./osx_environment.sh
-    B_CMAKE_FLAGS="-DCMAKE_OSX_SYSROOT=$(xcode-select --print-path)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 $B_CMAKE_FLAGS"
+    # macOS needs a little help, so we source this environment script to fix paths.
+    . ./macos_environment.sh
+
+    B_CMAKE_FLAGS="${B_CMAKE_FLAGS} -DCMAKE_OSX_SYSROOT=$(xcode-select --print-path)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9"
 fi
-# allow local customizations to build environment
-[ -r ./build_env.sh ] && . ./build_env.sh
+
+# Source local build variables to the environment, if available.
+# If not, continue as normal, and silently.
+[ -e "./build_env.sh" ] && . "./build_env.sh"
 
 # Initialise Git submodules
 git submodule update --init --recursive
 
-B_CMAKE_FLAGS="-DCMAKE_BUILD_TYPE=$B_BUILD_TYPE $B_CMAKE_FLAGS"
```

Comment:
B_CMAKE_FLAGS changes should live in a separate commit.

Reply:
This has been resolved in b0d6d989. You'll notice I also kept the `B_BUILD_TYPE` changes in the same commit. As I explained the commit message, my reasoning for this was because `B_CMAKE_FLAGS` relies on the `B_BUILD_TYPE` variable. I felt it was prudent to include both changes in the same commit, to avoid a breakage between the two separate commits that *could* have been made for the two changes. If you don't think that's a good idea, I can rollback...